ion_schema_2_0/constraints/regex.isl

```diff
@@ -0,0 +1,1091 @@
+    "@",
+    "🤨",
+  ],
+  should_reject_as_invalid: [
+    "1",
+    "2",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+  ]
+}
+
+type::{
+  name: regex_character_class_unioned_with_w_metachar_class,
+  regex: "[@\\w]",
+}
+$test::{
+  type: regex_character_class_unioned_with_w_metachar_class,
+  should_accept_as_valid: [
+    "0",
+    "1",
+    "2",
+    "3",
+    "a",
+    "b",
+    "c",
+    "D",
+    "E",
+    "F",
+    "_",
+    "@",
+  ],
+  should_reject_as_invalid: [
+    "}",
+    "π",
+    " ",
+    "\n",
+    "\t",
+    "\b",
+    ".",
+    "#",
+    "🤨",
+  ]
+}
+
+type::{
+  name: regex_character_class_unioned_with_W_metachar_class,
+  regex: "[abc\\W]",
+}
+$test::{
+  type: regex_character_class_unioned_with_W_metachar_class,
+  should_accept_as_valid: [
+    "a",
+    "b",
+    "c",
+    "[",
+    "}",
+    "π",
+    " ",
+    "\n",
+    ".",
+    "@",
+    "🤨",
+  ],
+  should_reject_as_invalid: [
+    "0",
+    "1",
+    "2",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+    "d",
+    "e",
+    "f",
+    "A",
+    "B",
+    "C",
+    "_",
+  ]
+}
+
+type::{
+  name: regex_character_class_unioned_with_s_metachar_class,
+  regex: "[abc\\s]",
+}
+$test::{
+  type: regex_character_class_unioned_with_s_metachar_class,
+  should_accept_as_valid: [
+    "a",
+    "b",
+    "c",
+    " ",
+    "\f",
+    "\r",
+    "\t",
+    "\n",
+  ],
+  should_reject_as_invalid: [
+    "0",
+    "1",
+    "2",
+    "3",
+    "}",
+    "π",
+    ".",
+    "@",
+    "🤨",
+    "\b",
+  ]
+}
+
+type::{
+  name: regex_character_class_unioned_with_S_metachar_class,
+  regex: "[ \\S]",
+}
+$test::{
+  type: regex_character_class_unioned_with_S_metachar_class,
+  should_accept_as_valid: [
+    "0",
+    "a",
+    "B",
+    "[",
+    "}",
+    "π",
+    ".",
+    "@",
+    "🤨",
+    "\b",
+    " ",
+  ],
+  should_reject_as_invalid: [
+    "\f",
+    "\n",
+    "\r",
+    "\t",
+  ]
+}
+
+type::{
+  name: regex_quantifier_at_least_n,
+  regex: "ab{3,}c",
+}
+$test::{
+  type: regex_quantifier_at_least_n,
+  should_accept_as_valid: [
+    "abbbc",
+    "abbbbc",
+    "abbbbbc",
+  ],
+  should_reject_as_invalid: [
+    "ac",
+    "abc",
+    "abbc",
+  ]
+}
+
+type::{
+  name: regex_quantifier_exactly_n,
+  regex: "ab{3}c",
+}
+$test::{
+  type: regex_quantifier_exactly_n,
+  should_accept_as_valid: [
+    "abbbc",
+  ],
+  should_reject_as_invalid: [
+    "abbc",
+    "abbbbc",
+  ]
+}
+
+type::{
+  name: regex_quantifier_one_to_many,
+  regex: "ab+c",
+}
+$test::{
+  type: regex_quantifier_one_to_many,
+  should_accept_as_valid: [
+    "abc",
+    "abbc",
+    "abbbc",
+  ],
+  should_reject_as_invalid: [
+    "ac",
+  ]
+}
+
+type::{
+  name: regex_quantifier_range,
+  regex: "ab{3,6}c",
+}
+$test::{
+  type: regex_quantifier_range,
+  should_accept_as_valid: [
+    "abbbc",
+    "abbbbc",
+    "abbbbbc",
+    "abbbbbbc",
+  ],
+  should_reject_as_invalid: [
+    "abbc",
+    "abbbbbbbc",
+  ]
+}
```

Comment:
```suggestion
type::{
  name: regex_quantifier_range,
  regex: "ab{2,4}c",
}
$test::{
  type: regex_quantifier_range,
  should_accept_as_valid: [
    "abbc",
    "abbbc",
    "abbbc",
  ],
  should_reject_as_invalid: [
    "abc",
    "abbbbbc",
  ]
}
```

Shrinking this makes it easier to count at a glance, fwiw. Still demonstrates the span including a non-referenced number (3 is between [2, 4]) if that's what we're going for here.